### ISSUE TITLE
The affinity-assistant pods are still spawning after disabling the associated feature-flag: disable-affinity-assistant

### ISSUE THREAD

**Issue by lmserrano** [tektoncd/pipeline#2979]
# Expected Behavior

No affinity-assistant pods should spawn

# Actual Behavior

There is an affinity-assistant pod spawning for each pipelinerun task that manages to execute

# Steps to Reproduce the Problem

1. Install Tekton Pipeline 0.14.2
2. `kubectl edit configmap feature-flags -n tekton-pipelines` and add `disable-affinity-assistant` as follows:
    ```
    apiVersion: v1
    data:
      disable-affinity-assistant: "true"
    ...
    ```
3. Start a pipelinerun which uses multiple persistent volume claims (the one i'm running specifically uses 1 pvc template and 2 pre-existing claims, but I don't think that should make a difference)

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.6-eks-4e7f64", GitCommit:"4e7f642f9f4cbb3c39a4fc6ee84fe341a8ade94c", GitTreeState:"clean", BuildDate:"2020-06-11T13:55:35Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

- Tekton Pipeline version:

Tekton Pipeline: v0.14.2

**Comment by shimmerjs** [comment]
Is real and is really annoying

**Comment by afrittoli** [comment]
The affinity assistant code tries to read the config value from context or default
https://github.com/tektoncd/pipeline/blob/abb677dbb23cdc305b0c73409679af980fa322c4/pkg/reconciler/pipelinerun/affinity_assistant.go#L223-L226

The pipelinerun controller has its own config store, which does not include feature flags, so the code above will always return the default. This should be indirectly solved by: https://github.com/tektoncd/pipeline/pull/2938 :)

**Comment by jlpettersson** [comment]
Seem to been introduced / changed in https://github.com/tektoncd/pipeline/pull/2637

**Comment by vdemeester** [comment]
This should be fixed in 0.14.3 and in master with :
- https://github.com/tektoncd/pipeline/pull/3015 (0.14.3)
- https://github.com/tektoncd/pipeline/pull/2938 (master)

I think we can "safely" close this. If this turns out to still be a bug, please comment here and I'll re-open :angel:

**Comment by abayer** [comment]
I seem to still be seeing this with 0.14.3...
